lib/pages/home/browse.dart

```diff
@@ -168,18 +169,59 @@ class _BrowsePageState extends State<BrowsePage> {
           ],
         ),
       ),
-      floatingActionButton: FloatingActionButton(
-        shape: cupertino ? const CircleBorder() : null,
-        onPressed: () async {
-          final newFilePath = await FileManager.newFilePath('${path ?? ''}/');
-          if (!mounted) return;
-          context.push(RoutePaths.editFilePath(newFilePath));
+      floatingActionButton: SpeedDial(
```

Comment:
Please can you make this a separate widget so the code isn't duplicated in browse.dart and recent_notes.dart?